builtin/providers/azurerm/resource_arm_local_network_gateway.go

```diff
@@ -0,0 +1,158 @@
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": &schema.Schema{
+				Type:      schema.TypeString,
+				Optional:  true,
+				ForceNew:  true,
+				StateFunc: azureRMNormalizeLocation,
+			},
+
+			"resource_group_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"resource_guid": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"gateway_address": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"address_space": &schema.Schema{
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+// resourceArmLocalNetworkGatewayCreate goes ahead and creates the specified ARM local network gateway.
+func resourceArmLocalNetworkGatewayCreate(d *schema.ResourceData, meta interface{}) error {
+	lnetClient := meta.(*ArmClient).localNetConnClient
+
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	ipAddress := d.Get("gateway_address").(string)
+
+	// NOTE: due to the including-but-different relationship between the ASM
+	// and ARM APIs, one may set the following local network gateway type to
+	// "Classic" and basically get an old ASM local network connection through
+	// the ARM API. This functionality is redundant with respect to the old
+	// ASM-based implementation which we already have, so we just use the
+	// new Resource Manager APIs here:
+	typ := "Resource Manager"
+
+	// fetch the 'address_space_prefix'es:
+	prefixes := []string{}
+	for _, pref := range d.Get("addres_space").([]interface{}) {
+		prefixes = append(prefixes, pref.(string))
+	}
+
+	// NOTE: result ignored here; review below...
+	resp, err := lnetClient.CreateOrUpdate(resGroup, name, network.LocalNetworkGateway{
+		Name:     &name,
+		Location: &location,
+		Type:     &typ,
+		Properties: &network.LocalNetworkGatewayPropertiesFormat{
+			LocalNetworkAddressSpace: &network.AddressSpace{
+				AddressPrefixes: &prefixes,
+			},
+			GatewayIPAddress: &ipAddress,
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("Error reading the state of Azure ARM Local Network Gateway '%s': %s", name, err)
+	}
+
+	// NOTE: we either call read here or basically repeat the reading process
+	// with the ignored network.LocalNetworkGateway result of the above:
+	d.SetId(*resp.ID)
+	return resourceArmLocalNetworkGatewayRead(d, meta)
+}
+
+// resourceArmLocalNetworkGatewayRead goes ahead and reads the state of the corresponding ARM local network gateway.
+func resourceArmLocalNetworkGatewayRead(d *schema.ResourceData, meta interface{}) error {
+	lnetClient := meta.(*ArmClient).localNetConnClient
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
```

Comment:
The correct pattern here as per the exemplar resources is to use `parseAzureResourceID` as per [virtual network](https://github.com/hashicorp/terraform/blob/master/builtin/providers/azurerm/resource_arm_virtual_network.go#L121)